### PR TITLE
rfc-10, serialization

### PR DESCRIPTION
**Pull request by piogor**

### CODE REVIEW

File: 10/RequestReply.md [hunk 1]
```diff
@@ -83,7 +83,37 @@ TODO understand the PUSH_BACK model. CORBA or more generic?
 
 #### Cache
 
-#### Serialisation
+#### Serialization
+
+The Serialization Mode defines how the Device Server handles multiple requests. 
+
+The Serialization SHALL apply to all requests independently of a request origin (network, the same machine, the same Device Server process).
```
Comment (lorenzopivetta) [inline 10/RequestReply.md]: Suggest to rephrase as:
s/network, the same machine/any network host, localhost/

File: 10/RequestReply.md [hunk 2]
```diff
@@ -83,7 +83,37 @@ TODO understand the PUSH_BACK model. CORBA or more generic?
...
+
+The Serialization SHALL apply to all requests independently of a request origin (network, the same machine, the same Device Server process). 
+
+A Device Server MAY handle requests with the following Serialization Modes:
+* Serialization by Device
+* Serialization by Class
+* Serialization by Process
+* No Serialization
+
+If Device Server is set to Serialization by Device:
```
Comment (lorenzopivetta) [inline 10/RequestReply.md]: s/If a Device/When a Device/

File: 10/RequestReply.md [hunk 3]
```diff
@@ -83,7 +83,37 @@ TODO understand the PUSH_BACK model. CORBA or more generic?
...
+* it SHALL process requests to each Device in the order of requests arrival,
+* for any of Devices, it MUST NOT start the processing of a new request until the processing of the previous request is not finished,
```
Comment (lorenzopivetta) [inline 10/RequestReply.md]: s/for any of Devices/for any Device in the Device Server/

File: 10/RequestReply.md [hunk 4]
```diff
@@ -83,7 +83,37 @@ TODO understand the PUSH_BACK model. CORBA or more generic?
...
+* it MAY process multiple requests in parallel, providing that there is maximum one processed request per a Device.
```
Comment (lorenzopivetta) [inline 10/RequestReply.md:101]: Not sure we use MAY here.. .as the serialization happens at Device level, so it MUST be in parallel for all the Devices

File: 10/RequestReply.md [hunk 5]
```diff
@@ -83,7 +83,37 @@ TODO understand the PUSH_BACK model. CORBA or more generic?
...
+
+If Device Server is set to Serialization by Class:
```
Comment (lorenzopivetta) [inline 10/RequestReply.md]: s/If a Device/When a Device/

File: 10/RequestReply.md [hunk 6]
```diff
@@ -83,7 +83,37 @@ TODO understand the PUSH_BACK model. CORBA or more generic?
...
+* it SHALL process requests to each set of Devices belonging to the same Class in the order of requests arrival,
+* for each Device Class, it MUST NOT start the processing of a new request until the processing of the previous request is not finished,
+* it MAY process multiple requests in parallel, providing that there is maximum one processed request per Device Class.
```
Comment (lorenzopivetta) [inline 10/RequestReply.md:106]: Same consideration here... should't this read MUST ?

File: 10/RequestReply.md [hunk 7]
```diff
@@ -83,7 +83,37 @@ TODO understand the PUSH_BACK model. CORBA or more generic?
...
+
+If Device Server is set to Serialization by Process:
+* it SHALL process requests in the order of requests arrival,
+* it MUST NOT start the processing of a new request until the processing of the previous request is not finished,
+
+The Device Server SHOULD process requests as soon as possible. If an incoming request cannot be processed due to Device Server 
+Serialisation Mode it SHOULD be queued for later processing and served as soon as possible.
+
+If Device Server is set to No Serialization it MAY process any requests in parallel. 
+It is NOT RECOMMENDED to use No Serialization mode.
```
Comment (lorenzopivetta) [inline 10/RequestReply.md:116]: I'd rephrase this, adding some explanation... what about:
"It is NOT RECOMMENDED to use No Serialization mode, unless the developer takes care of managing any concurrent access to the Device data structures." or something similar...

Comment (lorenzopivetta) [reply 10/RequestReply.md:116]: I'd also add:
 The default serialization Mode is "Serialization by Device".

**Review by lorenzopivetta — COMMENTED** [review]
I've added some comments

File: 10/RequestReply.md (hunk 1 above)
Comment (piogor) [reply 10/RequestReply.md]:
```suggestion
The Serialization SHALL apply to all requests independently of a request origin (any network host, localhost, the same Device Server process). 
```

File: 10/RequestReply.md (hunk 3 above)
Comment (piogor) [reply 10/RequestReply.md]:
```suggestion
* for any Device in the Device Server, it MUST NOT start the processing of a new request until the processing of the previous request is not finished,
```

File: 10/RequestReply.md (hunk 2 above)
Comment (piogor) [reply 10/RequestReply.md]:
```suggestion
When a Device Server is set to Serialization by Device:
```

File: 10/RequestReply.md (hunk 4 above)
Comment (piogor) [reply 10/RequestReply.md:101]: I think `MUST` would be to strong, as the parallel processing is controlled by an operating system and it is up to its scheduler.

File: 10/RequestReply.md (hunk 5 above)
Comment (piogor) [reply 10/RequestReply.md]:
```suggestion
When a Device Server is set to Serialization by Class:
```

File: 10/RequestReply.md (hunk 7 above)
Comment (piogor) [reply 10/RequestReply.md:116]:
```suggestion
It is NOT RECOMMENDED to use No Serialization mode. 

The default serialization Mode is "Serialization by Device".
```

**Comment by piogor** [comment]
Thanks @lorenzopivetta, I have just applied most of your suggestions, except this for `MUST`, see [the comment](https://github.com/tango-controls/rfc/pull/106#discussion_r478345350) .
